src/planner/ompl/CRRT.cpp

```diff
@@ -165,83 +198,102 @@ void CRRT::freeMemory(void) {
     si_->copyState(motion->state, st);
     mStartTree->add(motion);
   }
 
-  if (mStartTree->size() == 0) {
+  if (mStartTree->size() == 0)
+  {
     return ::ompl::base::PlannerStatus::INVALID_START;
   }
 
   if (!mSampler)
     mSampler = si_->allocStateSampler();
 
-  Motion *solution = nullptr;
-  Motion *approxsol = nullptr;
+  Motion* solution = nullptr;
+  Motion* approxsol = nullptr;
   double approxdif = std::numeric_limits<double>::infinity();
-  Motion *rmotion = new Motion(si_);
-  ::ompl::base::State *rstate = rmotion->state;
-  ::ompl::base::State *xstate = si_->allocState(); /* temp state */
+  Motion* rmotion = new Motion(si_);
+  ::ompl::base::State* rstate = rmotion->state;
+  ::ompl::base::State* xstate = si_->allocState(); /* temp state */
 
   bool foundgoal = false;
-  while (_ptc == false) {
+  while (_ptc == false)
+  {
     /* sample random state (with goal biasing) */
-    if (goalSampleable && mRng.uniform01() < mGoalBias &&
-        goalSampleable->canSample())
+    if (goalSampleable && mRng.uniform01() < mGoalBias
+        && goalSampleable->canSample())
       goalSampleable->sampleGoal(rstate);
     else
       mSampler->sampleUniform(rstate);
 
     // Continue on invalid sample
-    if (!si_->isValid(rstate)) {
+    if (!si_->isValid(rstate))
+    {
       continue;
     }
 
     /* find closest state in the tree */
-    Motion *nmotion = mStartTree->nearest(rmotion);
+    Motion* nmotion = mStartTree->nearest(rmotion);
 
     /* Perform a constrained extension */
     double bestdist = std::numeric_limits<double>::infinity();
-    Motion *bestmotion = constrainedExtend(_ptc, mStartTree, nmotion, rmotion->state, xstate, goal,
-                                           false, bestdist, foundgoal);
-    if (foundgoal) {
+    Motion* bestmotion = constrainedExtend(
+        _ptc,
+        mStartTree,
+        nmotion,
+        rmotion->state,
+        xstate,
+        goal,
+        false,
+        bestdist,
+        foundgoal);
+    if (foundgoal)
+    {
       solution = bestmotion;
       break;
-    } else if (bestdist < approxdif) {
+    }
+    else if (bestdist < approxdif)
+    {
       approxdif = bestdist;
       approxsol = bestmotion;
     }
   }
 
   bool solved = false;
   bool approximate = false;
-  if (solution == nullptr) {
+  if (solution == nullptr)
+  {
     solution = approxsol;
     approximate = true;
   }
 
-  if (solution != nullptr) {
+  if (solution != nullptr)
+  {
     mLastGoalMotion = solution;
 
     /* construct the solution path */
-    std::vector<Motion *> mpath;
-    while (solution != nullptr) {
+    std::vector<Motion*> mpath;
+    while (solution != nullptr)
+    {
       mpath.push_back(solution);
       solution = solution->parent;
     }
 
     /* set the solution path */
-    ::ompl::geometric::PathGeometric *path =
-        new ::ompl::geometric::PathGeometric(si_);
+    ::ompl::geometric::PathGeometric* path
+        = new ::ompl::geometric::PathGeometric(si_);
```

Comment:
This will leak memory if constructing the path raises an exception. We should create the `PathPtr` (or - if necessary - a `shared_ptr<PathGeometric>` here, instead of several lines below.